### DISCUSSION TITLE
Prioirty field for packet send

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy#1489] [Q&A] [answered]
hello @puddly ,

While I understand the concept of priority, I'm trying to figure out what could be the use case ?
Why should I send to a specific device a packet before and other one ?

**Comment by puddly** [comment, marked as answer]
We want to make sure that user interactions have the highest priority so that bad devices don't overload the network. If a device is asking for a default response, it will be sent after a user interaction with a device. If new devices are joining, we want those packets to be sent before anything else.

**Reply by pipiche38** [reply]
Thanks for the explanations.

On the Z4D plugin we are "virually" having a queue per device (based on IEEE). Due to that if a new device is coming it cannot be penalise by other requests on other devices.

**Reply by puddly** [reply]
This is also done now in zigpy. We limit devices to one concurrent request.
